source/common/res/features/split-transaction-auto-adjust/main.js

```diff
@@ -0,0 +1,109 @@
+      var isInitialized = false;
+      var addingAnotherSplit = false;
+      var deletingSplit = false;
+
+      function initialize() {
+        splitTransactionRow = $('.ynab-grid-add-rows');
+        addAnotherSplit = $('.ynab-grid-split-add-sub-transaction');
+
+        splitTransactionRow.on('keyup', '.currency-input .ember-text-field', onKeyPress);
+        splitTransactionRow.on('click', '.ynab-grid-sub-remove', onDeleteSplit);
+        addAnotherSplit.on('click', onAddAnotherSplit);
+      }
+
+      function onKeyPress() {
+        var element = this;
+        var currentInputClass = getCurrentInputClass();
+
+        if ($(element).parents(currentInputClass).length) {
+          autoFillNextRow(this);
+        }
+      }
+
+      function getCurrentInputClass() {
+        var firstRow = $('.ynab-grid-body-row', splitTransactionRow).first();
+        var outflowValue = parseFloat($('.ynab-grid-cell-outflow .ember-text-field', firstRow).val());
+        var inflowValue = parseFloat($('.ynab-grid-cell-inflow .ember-text-field', firstRow).val());
+        return outflowValue > 0 ? '.ynab-grid-cell-outflow' :
+               inflowValue > 0 ? '.ynab-grid-cell-inflow' : false;
+      }
+
+      function onAddAnotherSplit() {
+        addingAnotherSplit = true;
+      }
+
+      function onDeleteSplit() {
+        deletingSplit = true;
+      }
+
+      function autoFillNextRow(currentInputElement) {
+        var inputClass = getCurrentInputClass();
+        var total = parseFloat($(inputClass + ' .ember-text-field', splitTransactionRow.children().eq(0)).val());
+
+        if (inputClass && total) {
+          var currentRow = $(currentInputElement).parents('.ynab-grid-body-row');
+          var currentRowIndex = splitTransactionRow.children().index(currentRow);
+          var currentValue = parseFloat($(currentInputElement).val());
+
+          splitTransactionRow.children().each(function (index, splitRow) {
+            if (index === currentRowIndex) {
+              var nextRow = splitTransactionRow.children().eq(currentRowIndex + 1);
+              if (index === 0) {
+                $(inputClass + ' .ember-text-field', nextRow).val(total.toFixed(2));
+                $(inputClass + ' .ember-text-field', nextRow).trigger('change');
+              } else {
+                total -= currentValue;
+                $(inputClass + ' .ember-text-field', nextRow).val(total.toFixed(2));
+                $(inputClass + ' .ember-text-field', nextRow).trigger('change');
+              }
+            } else if (index < currentRowIndex) {
+              if (index !== 0) { // don't decrement total if we're the total row, that's silly
+                total -= parseFloat($(inputClass + ' .ember-text-field', splitRow).val());
+              }
+            }
+          });
+        }
+      }
+
+      return {
+        observe(changedNodes) {
+          var addTransactionSplit = changedNodes.regex(/modal-account-categories-split-transaction/i);
```

Comment:
Regex matching is a lot slower than doing a O(1) set key lookup. Is it necessary to use the regex matching method here? E.g. how long is the class string on these nodes, and could we just do an exact match for performance?